src/main/java/com/multiintstore/controllers/StorageController.java

```diff
@@ -15,13 +15,65 @@
  */
 package com.multiintstore.controllers;
 
-import org.springframework.web.bind.annotation.RequestMapping;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.multiintstore.callbacks.CallbackValidator;
+import com.multiintstore.callbacks.FinishedCallback;
+import com.multiintstore.callbacks.MetadataCallback;
+import com.multiintstore.callbacks.ProductCallback;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
-@RequestMapping("/store")
 public class StorageController {
```

Comment:
❓ Should we abstract the callback handling out of the controller so that we can keep it decoupled?

Reply:
Sure but I'll keep all the JsonNode stuff within that